### PR TITLE
Update documentation

### LINKED ISSUE #277: Update documentation
The documentation is a little outdated by now. This should be updated to match the current version.

### PR DESCRIPTION
**Pull request by haerdib**
closes #277

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,47 +14,56 @@ substrate-api-client a library written in Rust for connecting to the substrate's
...
+
+The following examples can be found in the [examples](/examples) folder:
+
+* [example_benchmark_bulk_xt](/examples/example_benchmark_bulk_xt.rs): Float the node with a series of transactions.
+* [example_compose_extrinsic_offline](/examples/example_compose_extrinsic_offline.rs): Compose an extrinsic without interacting with the node.
+* [example_contract_instantiate_with_code](/examples/example_contract_instantiate_with_code.rs): Instantiate a contract on the chain.
+* [example_contract](/examples/example_contract.rs): Handle ink! contracts (put, create, and call). **DEPRECATED!**
+* [example_custom_storage_struct](/examples/example_custom_storage_struct.rs): Fetch and decode custom structs from the runtime. **DEPRECATED!**
+* [example_event_callback](/examples/example_event_callback.rs): Subscribe and react on events.
+* [example_event_error_details](/examples/example_event_error_details.rs): Listen to error events from the node to determine if extrinsic was successful or not.
```
Comment (gaudenzkessler) [inline README.md:59]:
```suggestion
* [example_event_error_details](/examples/example_event_error_details.rs): Listen to error events from the node to determine if an extrinsic was successful or not.
```

File: README.md [hunk 2]
```diff
@@ -14,47 +14,56 @@ substrate-api-client a library written in Rust for connecting to the substrate's
...
+* [example_generic_event_callback](/examples/example_generic_event_callback.rs): Listen to an example event from the node.
+* [example_generic_extrinsic](/examples/example_generic_extrinsic.rs): Compose an extrinsic for any call in any module by supplying the module and call name as strings.
+* [example_get_block](/examples/example_get_block.rs): Read header, block and signed block from storage.
+* [example_get_storage](/examples/example_get_storage.rs): Read storage values.
+* [example_print_metadata](/examples/example_print_metadata.rs): Print the metadata of the node in a readable way.
+* [example_sudo](/examples/example_sudo.rs): Create and send a sudo wrapped call.
+* [example_transfer](/examples/example_transfer.rs): Transfer tokens by using a wrapper of compose_extrinsic
```
Comment (gaudenzkessler) [inline README.md:66]:
```suggestion
* [example_transfer](/examples/example_transfer.rs): Transfer tokens by using a wrapper of compose_extrinsic.
```

**Review by gaudenzkessler — APPROVED** [review]
Thank you very much! Only some tiny suggestions.

File: README.md [hunk 3]
```diff
@@ -14,47 +14,56 @@ substrate-api-client a library written in Rust for connecting to the substrate's
...
+git clone https://github.com/scs/substrate-api-client.git
+cd substrate-api-client
+cargo run --example example_get_storage
 ```
+or download the already built binaries from Github Actions: https://github.com/scs/substrate-api-client/actions and run them without any building:
 
-Set the output verbosity by prepending `RUST_LOG=info` or `RUST_LOG=debug`.
+```bash
+chmod +x EXAMPLE
+./EXAMPLE
```
Comment (clangenb) [inline README.md:45]: Is `EXAMPLE` a placeholder for the example of choice? If yes, you usually use brackets for placeholders `<example>` or `<EXAMPLE>`.

File: README.md [hunk 4]
```diff
@@ -14,47 +14,56 @@ substrate-api-client a library written in Rust for connecting to the substrate's
...
 In order to build the substrate-api-client and the examples, Rust and the wasm target are needed. For Linux:
-
-    curl https://sh.rustup.rs -sSf | sh
-
-    rustup default nightly
-    rustup target add wasm32-unknown-unknown --toolchain nightly
-
+```
+curl https://sh.rustup.rs -sSf | sh
+rustup show
```
Comment (clangenb) [inline README.md:19]: Maybe a comment that `rustup show` automatically installs the suitable toolchain would be nice.

**Review by clangenb — COMMENTED** [review]
Minor comments, but otherwise thanks for the maintenance!

File: README.md (hunk 3 above)
Comment (haerdib) [reply README.md]: Good input, fixed in [6e71aaa](https://github.com/scs/substrate-api-client/pull/305/commits/6e71aaa2732fefefa84e3d199813fcdf53cd26ae)

File: README.md (hunk 4 above)
Comment (haerdib) [reply README.md:20]: Better now?

File: rust-toolchain.toml [hunk 5]
```diff
@@ -1,4 +1,4 @@
 [toolchain]
-channel = "nightly-2022-08-18"
+channel = "nightly-2022-08-19"
```
Comment (clangenb) [inline rust-toolchain.toml]: Haha 🚀

**Review by clangenb — APPROVED** [review]
Nice, looks good to me!

File: rust-toolchain.toml (hunk 5 above)
Comment (haerdib) [reply rust-toolchain.toml]: Upsi

Comment (haerdib) [reply rust-toolchain.toml]: Yeah, somehow 08-18 doens't work for me. Never found out why. Fixed, sorry :P

**Pull request merged**